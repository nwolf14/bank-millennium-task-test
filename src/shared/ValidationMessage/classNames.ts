export const className = 'ValidationMessage';

export const classNames = {
  [className]: className,
  helperText: `${className}__helper-text`,
  errorText: `${className}__error-text`,
  errorIcon: `${className}__error-icon`,
};
