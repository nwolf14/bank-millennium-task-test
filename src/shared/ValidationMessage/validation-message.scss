@import '../../styles/colors/common.scss';
@import '../../styles/colors/input/common.scss';

.ValidationMessage {
  &__helper-text {
    color: $--cth-helper-text;

    &--dark {
      color: $--cth-helper-text--dark;
    }
  }

  &__error-text {
    align-items: center;
    color: $--cth-error;
    display: flex;
  }

  &__helper-text,
  &__error-text {
    font-size: 12px;
    margin: 0;
    margin-left: 5px;
    margin-top: 5px;
  }

  &__error-icon {
    margin-right: 5px;
  }
}
