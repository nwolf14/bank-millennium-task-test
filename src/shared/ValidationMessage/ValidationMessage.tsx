import classnames from 'classnames';
import { FC } from 'react';

// hooks
import { useTheme } from '../../hooks/useTheme/useTheme';

// others
import { ReactComponent as ErrorIcon } from '../../assets/icons/error.svg';
import {
  className as classNameValidationMessage,
  classNames,
} from './classNames';
import { TObject } from '../../types';

// styles
import './validation-message.scss';

export type TProps = {
  className?: string;
  error?: string;
  helperText?: string;
  style?: TObject<number | string>;
};

export const ValidationMessage: FC<TProps> = ({
  className = '',
  error,
  helperText,
  style = {},
}) => {
  const { classNamesWithTheme } = useTheme(classNames);

  return (
    <div
      className={classnames(
        classNamesWithTheme[classNameValidationMessage],
        className
      )}
      style={style}
    >
      {!error && helperText && (
        <p className={classNamesWithTheme.helperText}>{helperText}</p>
      )}
      {error && (
        <div className={classNamesWithTheme.errorText}>
          <ErrorIcon className={classNamesWithTheme.errorIcon} />
          {error}
        </div>
      )}
    </div>
  );
};

export default ValidationMessage;
