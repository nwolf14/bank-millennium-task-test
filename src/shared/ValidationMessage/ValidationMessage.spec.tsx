import { render } from '@testing-library/react';

// components
import ValidationMessage from './ValidationMessage';

describe('ValidationMessage', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<ValidationMessage />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ValidationMessage error="error" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ValidationMessage helperText="helperText" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <ValidationMessage error="error" helperText="helperText" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
