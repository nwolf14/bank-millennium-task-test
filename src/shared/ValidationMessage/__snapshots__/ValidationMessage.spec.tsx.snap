// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationMessage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ValidationMessage"
  />
</DocumentFragment>
`;

exports[`ValidationMessage matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="ValidationMessage"
  >
    <div
      class="ValidationMessage__error-text"
    >
      <svg
        class="ValidationMessage__error-icon"
      >
        error.svg
      </svg>
      error
    </div>
  </div>
</DocumentFragment>
`;

exports[`ValidationMessage matches snapshot 3`] = `
<DocumentFragment>
  <div
    class="ValidationMessage"
  >
    <p
      class="ValidationMessage__helper-text"
    >
      helperText
    </p>
  </div>
</DocumentFragment>
`;

exports[`ValidationMessage matches snapshot 4`] = `
<DocumentFragment>
  <div
    class="ValidationMessage"
  >
    <div
      class="ValidationMessage__error-text"
    >
      <svg
        class="ValidationMessage__error-icon"
      >
        error.svg
      </svg>
      error
    </div>
  </div>
</DocumentFragment>
`;
